### DISCUSSION TITLE
Accessing Unit Model Values

### DISCUSSION THREAD

**Discussion by NCJazzMan65** [IDAES/idaes-pse#663] [Q&A] [answered]
How do I access the value of a specific unit model variable to use in other calculations?  For instance, say my flowsheet has a number of pump unit models and I want to add up the work calculated for each pump to get the total work for my model.

**Comment by andrewlee94** [comment, marked as answer]
Using the pump work for as an example, you can do the following:

```
m = ConcreteModel()
m.fs = FlowsheetBlock()

m.fs.pump1 = Pump()

m.fs.pump1.mechanical_work.display()

# Add a total work to the flowsheet that adds work of two pumps
m.fs.total_work = Expression(expr=m.fs.pump1.mechanical_work[0] + m.fs.pump2.mechanical_work[0])
```

A few complications arise however:

1. Note that most variables are indexed by time - e.g. in the sum above you need to include a time index on the `mechanical_work` terms.
2. IDAES models are hierarchical, so some variables are actually buried a few levels deeper - the most important ones of these are the state variables and properties which are buried in StateBlocks inside the ControlVolumes. However, as long as you know the full path to the variable, you can use it however you want.
